backend/src/apiserver/server/pipeline_upload_server.go

```diff
@@ -63,7 +65,13 @@ func (s *PipelineUploadServer) UploadPipeline(w http.ResponseWriter, r *http.Req
 		s.writeErrorToResponse(w, http.StatusBadRequest, util.Wrap(err, "Invalid pipeline name."))
 		return
 	}
-	newPipeline, err := s.resourceManager.CreatePipeline(pipelineName, "", pipelineFile)
+	// We don't set a max length for pipeline description here, since in our DB the description type is longtext.
+	pipelineDescription, err := url.QueryUnescape(r.URL.Query().Get(DescriptionQueryStringKey))
```

Comment:
curious, does r.URL.Query().Get(...) return empty string when the query doesn't exist?

Reply:
If a request comes without description field, do this pass with empty description or report an error?
I think we should remain backward compatibility on this. Just double checking with you because I'm not familiar with this part.

Reply:
In fact, I'm not sure we should get description from the query parameter. Description should already be in the compiled yaml file like the following:
```
"apiVersion": |-
  argoproj.io/v1alpha1
"kind": |-
  Workflow
"metadata":
  "annotations":
    "pipelines.kubeflow.org/pipeline_spec": |-
      {"description": "A pipeline with two sequential steps.", "inputs": [{"default": "gs://ml-pipeline-playground/shakespeare1.txt", "name": "url"}], "name": "Sequential pipeline"}
  "generateName": |-
    sequential-pipeline-
"spec":
  "arguments":
```

Reply:
synced offline, decision:
1. Pipeline description is specified by UI/SDK
2. Pipeline version description can parse yaml content automatically